AzureIotDps.X/mqtt_packetPopulation/mqtt_iotprovisioning_packetPopulate.c

```diff
@@ -161,7 +163,22 @@ void MQTT_CLIENT_iotprovisioning_connect(char* deviceID)
     device_id = az_span_slice(device_id, 0, az_span_size(deviceID_parm));
    
     const az_span global_device_endpoint = AZ_SPAN_LITERAL_FROM_STR(CFG_MQTT_PROVISIONING_HOST);
-    const az_span id_scope = AZ_SPAN_LITERAL_FROM_STR(PROVISIONING_ID_SCOPE);
+
+	// Read the ID Scope from the secure element (e.g. ATECC608A)
+	atcab_read_bytes_zone(ATCA_ZONE_DATA, ATCA_SLOT_IDSCOPE, 0, atca_id_scope, sizeof(atca_id_scope));
+	debug_printInfo("DPS: ID Scope = %s", atca_id_scope);
+
+	const az_span scopeID_parm = az_span_create_from_str((char*)atca_id_scope);
+	az_span id_scope = AZ_SPAN_FROM_BUFFER(scope_id_buf);
```

Comment:
I don't understand what you are doing here. It looks like you are copying the same buffer around.

Reply:
why can't line 171 be 

id_scope = az_span_create_from_str((char*)atca_id_scope);